src/dmd/backend/dwarfdbginf.d

```diff
@@ -601,6 +587,24 @@ static if (1)
         public uint[TYMAX] typidx_tab;
     }
 
+    /*****************************************
+     * Replace the bytes in `buf` from the `offset` by `data`.
+     *
+     * Params:
+     *      buf = buffer where `data` will be written
+     *      offset = offset of the bytes in `buf` to replace
+     *      data = bytes to write
+     */
+    extern(D) void rewrite(T)(Outbuffer* buf, uint offset, T data)
+    {
+        ulong end = offset + data.sizeof;
+        for(uint i = 0; i < end - offset; ++i)
+            buf.buf[offset + i] = (data >> (8 * i)) & 0xff;
+    }
+
+    alias rewrite32 = rewrite!uint;
+    alias rewrite64 = rewrite!ulong;
```

Comment:
rewrite64 is never used, making the template pointless.

Reply:
Don't worry, it will be used later for the DWARF 64 bits format ;)

Reply:
Ok, but it also is a very very slow re-implementation of memcpy. It can be as simple as `*(T*)&buf = t;`, so simple that no template is needed. No need to worry about endianess, as dmd is x86 only.

Reply:
Ok, thanks for the infos. However, I don't like using `*(T*)&buf = t;` everywhere in the code, I think it's better to keep the template and rewrite it with your simple solution.